quai/filters/api.go

```diff
@@ -289,6 +289,63 @@ func (api *PublicFilterAPI) NewHeads(ctx context.Context) (*rpc.Subscription, er
 	return rpcSub, nil
 }
 
+// Accesses send a notification each time the specified address is accessed
+func (api *PublicFilterAPI) Accesses(ctx context.Context, addr common.Address) (*rpc.Subscription, error) {
```

Comment:
Will each subscription launch its own goroutine?
Seems like it would blow up if, on each block/transaction we're checking the same access list over and over for each address we're interested in, instead of doing it all at once.

Reply:
That's a fair point. This is how the entire subscription system works, so I was just copying the existing paradigm. That said, I think a nice optimization later would be to combine the different filters into one search loop.

Reply:
If not now could you please make the issue, could cause some major slowdowns depending on throughput and number of subs.

Reply:
we have a stateful filtersystem which can do this.

Reply:
well I don't want to do it now, because the easier "everyone use the same thread" approach would actually be slower for anything between 1-N subscriptions, where N is the number of threads available. With go's green threads, the tradeoff point could actually be quite high.

What really needs to be done, is to create a threadpool for this sort of activity and batch work once the threadpool is saturated. That, however, is a lot of work for an RPC performance optimization.

I've created an issue: https://github.com/dominant-strategies/go-quai/issues/2121